packages/did/src/Did.utils.ts

```diff
@@ -455,3 +466,20 @@ export function assembleKeyUri(
   }
   return `${did}#${keyId}`
 }
+
+/**
+ * Helper to produce polkadot-js style enum representations, consisting of an object with a single key-value pair.
+ * The enum variant becomes the object's key (first letter capitalized).
+ *
+ * @param variant The enum variant descriptor as a string.
+ * @param value The value associated with the variant.
+ * @returns `{ Variant: value }`.
+ */
+export function makeJsonEnum<K extends string, V>(
```

Comment:
I'll try to come up with something. TypedValue doesn't seem right though, because what it is is a representation of a Rust enum in Typescript, where you have a fixed number of variants, and each variant can have a different (statically typed) value.

Reply:
How about `TypedVariants` then?

Reply:
I like it!